### PR TITLE
Handling JSON in MongoDB

### LINKED ISSUE #2366: Handling JSON data in MongoDB
### Introduction paragraph (2-3 paragraphs):
According to developer.mozilla.org, JavaScript Object Notation (JSON) is a standard text-based format for representing structured data based on JavaScript object syntax.

It is commonly used for transmitting data in web applications. This process involves sending and receiving data between the server and the client. 

This article will focus on explaining how to handle JSON data in MongoDB.

### Key takeaways: 
1. Understanding JSON
2. Data representation in MongoDB
3. How to use JSON data in MongoDB

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by salvador02**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2366

### CODE REVIEW

**Comment by hectorkambow** [comment]
@salvador02 
Good afternoon and thank you for submitting your article for publishing. It has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀

Also FYI - I see you have made a few commits after creating the PR - please only create a PR when the article is fully polished and ready to be reviewed. 

Otherwise you can make all those changes and commits on your forked repo 👍  

Just an FYI to avoid potential closure of future articles. Thank you. 

cc @ahmadmardeni1 & @louisefindlay23

**Comment by hectorkambow** [comment]
@Ericgacoki

Will be reviewing this PR 👍

**Comment by hectorkambow** [comment]
plagiarism check done @Ericgacoki

**Comment by Ericgacoki** [comment]
Hello @salvador02, welcome to the `EngEd` program 🎊. Thank you for submitting your first article. I'll be reviewing it to correct grammatical and technical errors (if any) before being published.

With appreciation @hectorkambow.

File: content/articles/handling-json-in-mongodb/index.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
+### Introduction
+MongoDB is a high-performance, document-oriented NoSQL database which came to light in the mid-2000s.Unlike SQL-based databases, MongoDB uses documents (records) and collections (tables) to store non-relational data. MongoDB database has grown in its use because of its scalability and ease of use.
+
+JavaScript Object Notation (JSON) is a data exchange format solely based on JavaScript. JSON objects can easily get transferred from one system to another due to their compatibility with the majority of systems. JSON is simple and easy to use besides being independent of any language. For these reasons, JSON has a wide range of applications in programming and data representation.
+
+JSON can store various data types like strings, arrays, objects, and Binary JSON (BSON). What makes access to these data easy is the ability of various tools to export and import JSON documents to and from other systems using the various available tools as JSON makes use of key-value pair notation presumed to be the basic units of data.
+
+### Table of Contents
```
Comment (Ericgacoki) [inline content/articles/handling-json-in-mongodb/index.md:8]:
```suggestion
### Table of content
```
Only capitalize the first word not multiple words in the heading, unless it's a proper noun, an acronym or a keyword.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]
Awesome article @salvador02, you've covered a great topic 🚀. Please check the suggestions I've made below.

**Comment by Ericgacoki** [comment]
@salvador02 ping me when you're done so that we can proceed to `tech review`.

**Comment by salvador02** [comment]
@Ericgacoki , Thank you for the suggestions, I committed them. Kindly check out.

**Comment by salvador02** [comment]
@Ericgacoki , Thank you for the suggestions, I committed them. Kindly check out.

File: content/authors/simon-salva/index.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+title: Simon Salva
+type: authors
+github: https://github.com/salvador02
+---
+Simon is a a pssionate computer programmer interested in contributing in open source projects. He likes travelling and playing chess.
```
[the remark this answers is not in the document]
Comment (Ericgacoki) [reply content/authors/simon-salva/index.md]: > Regarding your author image, please rename it to `avatar.jpg` and ensure that it's not too large. Consider using [Pixlr](https://pixlr.com/e/) or an alternative software to compress/crop it to a suitable size, preferably `600 X 600`px.

@salvador02, seems like you forgot to fix your avatar. Please do so.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]
Hey @salvador02, be sure to go through the comments left below a suggestion before committing it to avoid missing out some points. Also, remember to `pull` changes before your next local `commit`.

**Comment by salvador02** [comment]
@Ericgacoki, kindly check out the corrections. If there is anything pending you can guide me on how to go through it. I will appreciate.

**Comment by Ericgacoki** [comment]
Good work @salvador02, the only remaining thing at the moment is about your author image. I'm afraid you missed my previous comments about it. However, rename it to `avatar.jpg` and compress/crop it to a suitable size, preferably below 500KB.

**Comment by salvador02** [comment]
Is there a preferable image size? or a suggest a suitable compression tool.

**Comment by Ericgacoki** [comment]
I'd recommend you to use [Pixlr](https://pixlr.com/e/) (though you're not limited to this). `600 X 600px` would be a good size for the avatar.

**Comment by Ericgacoki** [comment]
Hello @salvador02, this image is similar to the one you deleted. 4MB is too large for an avatar. If you're facing any trouble addressing this issue, feel free to reach out to me via Slack for further assistance. Thank you!

**Comment by salvador02** [comment]
@Ericgacoki  I think that one is a now okay

**Comment by Ericgacoki** [comment]
Yeah, that looks great🙂. We're now proceeding towards `tech review`. At this stage, your article will be reviewed for empirical truth and the work-ability of your code.

cc:@hectorkambow

**Review by Ericgacoki — APPROVED** [review]

**Review by Ericgacoki — CHANGES_REQUESTED** [review]

**Comment by salvador02** [comment]
@Ericgacoki done!

File: content/articles/handling-json-in-mongodb/index.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
+### Introduction
+MongoDB is a high-performance, document-oriented NoSQL database which came to light in the mid-2000s. Unlike SQL-based databases, MongoDB uses documents (records) and collections (tables) to store non-relational data. MongoDB has grown in its use due to the scalability and its ease of use.
```
Comment (Ericgacoki) [inline content/articles/handling-json-in-mongodb/index.md:2]:
```suggestion
---
layout: engineering-education
status: publish
published: true
url: /handling-json-in-mongodb/
title: Handling JSON in MongoDB
description: In this article, we will learn how to structure, import, export, and save JSON data in MongoDB.
author: simon-salva
date: 2021-07-13T00:00:00-00:20
topics: []
excerpt_separator: <!--more-->
images:
  - url: /engineering-education/handling-json-in-mongodb/hero.png
    alt: Handling JSON in MongoDB image
---

MongoDB is a high-performance, document-oriented NoSQL database which came to light in the mid-2000s. Unlike SQL-based databases, MongoDB uses documents (records) and collections (tables) to store non-relational data.
<!--more-->
MongoDB has grown in its use due to the scalability and its ease of use.
```
Added front matter.

File: content/articles/handling-json-in-mongodb/index.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+### Conclusion
+In this article, we've learned how to handle JSON data in MongoDB and the various operations that can be performed on JSON using MongoDB. Among the operations are structuring data, importing and exporting data from JSON to MongoDB, and vice versa.
+You can find the JSON files [here.](https://github.com/salvador02/jsonfile)
+
+### Further Reading
+Now that you have an understanding of how to use JSON in MongoDB, you can check out these links to find out more on this topic:
+- [JSON and BSON](https://www.mongodb.com/json-and-bson)
+- [Mongoexport](https://docs.mongodb.com/manual/reference/program/mongoexport/)
+- [Mongoimport](https://docs.mongodb.com/manual/reference/program/mongoimport/)
```
Comment (Ericgacoki) [inline content/articles/handling-json-in-mongodb/index.md:205]:
```suggestion
### Further Reading
Now that you have an understanding of how to use JSON in MongoDB, you can check out these links to find out more on this topic:
- [JSON and BSON](https://www.mongodb.com/json-and-bson)
- [Mongoexport](https://docs.mongodb.com/manual/reference/program/mongoexport/)
- [Mongoimport](https://docs.mongodb.com/manual/reference/program/mongoimport/)

Happy coding!

---

Peer Review Contributions by: [Eric gacoki](/engineering-education/authors/eric-gacoki/)
```
Added peer reviewer line.

File: content/articles/handling-json-in-mongodb/index.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+Now let us export the new file to JSON using the following command:
+```shell
+mongoexport --db studentinfo --collection student --out d:\out.json --pretty
+```
+
+You should have the response below in your terminal:
+```shell
+2021-06-29T23:02:50.046+0300    connected to: mongodb://localhost/
+2021-06-29T23:02:50.071+0300    exported 4 records
+```
```
Comment (Ericgacoki) [inline content/articles/handling-json-in-mongodb/index.md:155]: Did you forget to change these ones? Please change `shell` to `bash`.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]
Hello @salvador02, I've completed `tech-review` and I'm impressed with your clean and well formatted code, keep up!🚀 Kindly go though the following suggestions before we head to `final review`.

File: content/articles/handling-json-in-mongodb/index.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+If your import was a success, you should get the message below in the terminal
+```bash
+2021-06-29T22:35:18.454+0300    connected to: mongodb://localhost/
+2021-06-29T22:35:18.500+0300    4 document(s) imported successfully. 0 document(s) failed to import.
+```
+
+Now to verify the import, run the command below to view the imported data:
+```bash
+db.students.find().pretty()
+```
```
Comment (Ericgacoki) [inline content/articles/handling-json-in-mongodb/index.md:112]:
```suggestion
```bash
db.students.find()
```
```

**Review by Ericgacoki — CHANGES_REQUESTED** [review]

File: content/articles/handling-json-in-mongodb/index.md [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+If your import was a success, you should get the message below in the terminal
+```bash
+2021-06-29T22:35:18.454+0300    connected to: mongodb://localhost/
+2021-06-29T22:35:18.500+0300    4 document(s) imported successfully. 0 document(s) failed to import.
+```
+
+Now to verify the import, run the command below to view the imported data:
+```bash
+db.students.find()
```
Comment (Ericgacoki) [inline content/articles/handling-json-in-mongodb/index.md:128]: @salvador02,
Seems like I mistakenly removed the closing back-ticks (```) in my previous suggestion. Kindly fix this and the article will be ready for `final review` 😎.

**Comment by salvador02** [comment]
@Ericgacoki  done!

**Review by Ericgacoki — APPROVED** [review]

**Comment by Ericgacoki** [comment]
Greetings @hectorkambow, this article is `ready for final review` 🚀.

**Comment by hectorkambow** [comment]
@salvador02 
Overall great work gents 

Please be sure to go over any edits made to improve future articles/reviews 🚀 


cc @Ericgacoki

**Pull request merged**